Etabs_Adapter/CRUD/Read/Results/MeshResults.cs

```diff
@@ -508,14 +698,48 @@ private List<MeshStress> SmoothenStresses(List<MeshStress> forces)
                 double pr2 = group.Average(x => x.Principal_2);
                 double pr1_2 = group.Average(x => x.Principal_1_2);
 
-                smoothenedForces.Add(new MeshStress(first.ObjectId, first.NodeId, "", first.ResultCase, first.ModeNumber, first.TimeStep, first.MeshResultLayer, first.LayerPosition, MeshResultSmoothingType.ByPanel, first.Orientation, sxx, syy, sxy, txx, tyy, pr1, pr2, pr1_2));
+                smoothenedForces.Add(new MeshStress(first.ObjectId, first.NodeId, "", first.ResultCase, first.ModeNumber, first.TimeStep, 
+                                                        first.MeshResultLayer, first.LayerPosition, MeshResultSmoothingType.ByPanel, first.Orientation, 
+                                                        sxx, syy, sxy, txx, tyy, pr1, pr2, pr1_2));
             }
 
             return smoothenedForces;
         }
 
         /***************************************************/
 
+
+        private List<MeshVonMises> SmoothenVonMisesStresses(List<MeshVonMises> forces)
+        {
+            List<MeshVonMises> smoothenedVMStresses = new List<MeshVonMises>();
+
+            foreach (IEnumerable<MeshVonMises> group in forces.GroupBy(x => new { x.ResultCase, x.TimeStep, x.NodeId }))
+            {
+                MeshVonMises first = group.First();
+
+                double s = group.Average(x => x.S);
+                double n = group.Average(x => x.N);
+                double m = group.Average(x => x.M);
```

Comment:
If you remove the vonMises moment as commented above, this is going to be an average of `double.NaN` - you may need to add some checks in here.

Reply:
@peterjamesnugent , removed the von mises moment. x.M=NaN in all instances now ;)